src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -273,6 +270,33 @@ def _validate_pulses(self, pulses: List[HardwarePulse]):
                 f"{[pulses[0].wires]} of all wires [{self.wires}]"
             )
 
+    def _validate_measurement_basis(self, observable):
+        """Confirm that all elements of the observable are in the measurement basis,
+        and otherwise raise an error"""
+
+        # if the observable is a composite of other operations,
+        # loop through those and evaluate individually
+        if isinstance(observable, CompositeOp):
+            for op in observable.operands:
+                self._validate_measurement_basis(op)
```

Comment:
Probably worth adding a case for when the observable is `SymbolicOp` for `s_prod`, etc.

Reply:
Currently left out because `SymbolicOp` has a single base operator, so we don't need to loop through the individual base operators like for `CompositeOp` or `Hamiltonian`. They can follow the same logic as any 'single operator' operation. Both `s_prod` and `exp` are included in the tests :)